### PR TITLE
fs: add .h file for fs automount feature

### PR DESCRIPTION
**Pull request by olegartys**
For autmount support exists implementation in
fs/mount/fs_automount.c, but the header with
declarations was missing. Copied it from nuttx.

Signed-off-by: Oleg Lyovin <o.lyovin@partner.samsung.com>

### CODE REVIEW

File: os/include/tinyara/fs/automount.h [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H
+#define __INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H
```
Comment (lokeshbv) [inline os/include/tinyara/fs/automount.h]: Change  `__INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H` to  `__INCLUDE_AUDIO_AUTOMOUNT_H`

File: os/include/tinyara/fs/automount.h [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+ */
+
+/* Helper macros ************************************************************/
+
+#define AUTOMOUNT_ATTACH(s,isr,arg) ((s)->attach(s,isr,arg))
+#define AUTOMOUNT_DETACH(s)         ((s)->attach(s,NULL,NULL))
+#define AUTOMOUNT_ENABLE(s)         ((s)->enable(s,true))
+#define AUTOMOUNT_DISABLE(s)        ((s)->enable(s,false))
+#define AUTOMOUNT_INSERTED(s)       ((s)->inserted(s))
+
```
Comment (lokeshbv) [inline os/include/tinyara/fs/automount.h]: Please add a space after comma in line 82 to 86

File: os/include/tinyara/fs/automount.h [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+void automount_uninitialize(FAR void *handle);
+
+#undef EXTERN
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* CONFIG_FS_AUTOMOUNTER */
+#endif /* __INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H */
```
Comment (lokeshbv) [inline os/include/tinyara/fs/automount.h]: Change  `__INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H` to  `__INCLUDE_AUDIO_AUTOMOUNT_H`

File: os/include/tinyara/fs/automount.h (hunk 1 above)
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: No, it should be __INCLUDE_TINYARA_FS_AUTOMOUNT_H.

File: os/include/tinyara/fs/automount.h [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <tinyara/irq.h>
+
+#ifdef CONFIG_FS_AUTOMOUNTER
```
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: You should include <tinyara/config.h> to use config.

Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: And, we don't have FS_AUTOMOUNTER in Kconfig. It should be added.

File: os/include/tinyara/fs/automount.h [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+/* This is the type of the automount media change handler.  The lower level
+ * code will intercept the interrupt and provide the upper level with the
+ * private data that was provided when the interrupt was attached and will
+ * also provide an indication if the media was inserted or removed.
+ */
+
+struct automount_lower_s; /* Forward reference.  Defined below */
+
+typedef CODE int
+  (*automount_handler_t)(FAR const struct automount_lower_s *lower,
```
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: There are many violations of coding rule.

Comment (olegartys) [inline os/include/tinyara/fs/automount.h]: @sunghan-chang Hello,

I have just copied the header from nuttx, is it OK to modify it's original coding style?

Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: Sure, you should follow our coding style if you want to add something in Tizen RT.

Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: Except license sentence, you can modify whole things.

**Comment by olegartys** [comment]
@sunghan-chang Please, review the latest patchset

File: os/include/tinyara/fs/automount.h [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_TINYARA_AUDIO_AUTOMOUNT_H
```
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: This is not an AUDIO. Let's change it to FS.
```
#ifndef __INCLUDE_TINYARA_FS_AUTOMOUNT_H
#define __INCLUDE_TINYARA_FS_AUTOMOUNT_H
```

File: os/include/tinyara/fs/automount.h [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+void automount_uninitialize(FAR void *handle);
+
+#undef EXTERN
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* CONFIG_FS_AUTOMOUNTER */
+#endif /* __INCLUDE_TINYARA_AUDIO_AUTOMOUNT_H */
```
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: same as previous

File: os/include/tinyara/fs/automount.h [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+ * Input Parameters:
+ *   lower - Persistent board configuration data
+ *
+ * Returned Value:
+ *   A void* handle.  The only use for this handle is with automount_uninitialize().
+ *   NULL is returned on any failure.
+ *
+ ****************************************************************************/
+
+FAR void *automount_initialize(FAR const struct automount_lower_s *lower);
```
Comment (sunghan-chang) [inline os/include/tinyara/fs/automount.h]: We don't have fs_automount.c, Nuttx has.
Why do we (you) need this functionality?

**Comment by olegartys** [comment]
As automount feature is not needed in open-source branch currently, I am closing pull-request

**Pull request closed without merging**